docs/ip_reuse.md

```diff
@@ -0,0 +1,53 @@
+# IP reuse
+
+## What is IP reuse
+
+As a developer or user there could be a need to make IP addresses or IP allocations that will be attached to the nodes in the chain of: **BMH <=> M3M <=> CAPI machine <=> node** objects predictable, for example during the upgrade process and the main goal of this feature is to make that possible.
+
+## Logic behind
+
+As is now, M3IPPool is an object representing a set of IP address pools to be used for IP address allocations. An M3IPClaim is an object representing a request for an IP address allocation. Consequently, the M3IPClaim name is structured as following:
+
+**M3IPClaimName** = **M3DataName** + **(-)** + **M3IPPoolName**
+
+Example: m3datatemplate-0-pool0
+
+First part in `M3IPClaimName`, `M3DataName` is always created in a format of `M3DataTemplateName-index` and second part is `M3IPPoolName` which is the name of Metal3IPPool object defined in the [IP Address manager repo](https://github.com/metal3-io/ip-address-manager).
```

Comment:
This part was a bit hard to understand I think. Tried to make it a bit easier in this suggestion:
```suggestion
The `M3DataName` is derived from the `M3DataTemplateName` with an added index (`M3DataTemplateName-index`), and the `M3IPPoolName` comes from the Metal3IPPool object directly. (See the [IP Address manager repo](https://github.com/metal3-io/ip-address-manager) for more details on these objects).
```

Reply:
Sure, updated